drivers/adc/adc_ads1x1x.c

```diff
@@ -138,11 +154,41 @@ struct ads1x1x_data {
 	int16_t *buffer;
 	int16_t *repeat_buffer;
 	struct k_thread thread;
+	k_tid_t	tid;
 	bool differential;
+#ifdef ADC_ADS1X1X_TRIGGER
+	struct gpio_callback gpio_cb;
+	struct k_work work;
+#endif
 
 	K_THREAD_STACK_MEMBER(stack, CONFIG_ADC_ADS1X1X_ACQUISITION_THREAD_STACK_SIZE);
 };
 
+#ifdef ADC_ADS1X1X_TRIGGER
+static inline void ads1x1x_setup_rdy_pin(const struct device *dev, bool enable)
+{
+	const struct ads1x1x_config *config = dev->config;
+	gpio_flags_t flags = enable
+		? GPIO_INPUT | config->alert_rdy.dt_flags
+		: GPIO_DISCONNECTED;
+
+	gpio_pin_configure(config->alert_rdy.port,
+			   config->alert_rdy.pin,
+			   flags);
+}
+
+static inline void ads1x1x_setup_rdy_interrupt(const struct device *dev, bool enable)
+{
+	const struct ads1x1x_config *config = dev->config;
+	gpio_flags_t flags = enable
+		? GPIO_INT_EDGE_FALLING
+		: GPIO_INT_DISABLE;
+
+	gpio_pin_interrupt_configure(config->alert_rdy.port, config->alert_rdy.pin,
+				     flags);
```

Comment:
`gpio_pin_interrupt_configure_dt()`  instead? and also the return value is being ignored.